src/App.jsx

```diff
@@ -3,24 +3,15 @@ import './App.css'
 import Layout from './views/Layout/Layout'
 import Home from './views/Home/Home'
 import fetchUser from './services/user'
+import { useUser } from './context/UserContext'
 
 function App() {
   // inital value should match the data type of end value
-  const [user, setUser] = useState({})
-
-  useEffect(() => {
-    fetchUser()
-      .then((fetchedUser) => {
-        setUser(fetchedUser)
-      })
-      .catch((error) => {
-        throw new Error(`Error: ${error}`)
-      })
-  }, [])
+  const { setUser } = useUser()
```

Comment:
Since you moved the useEffect into the context provider, you don't need this here.